src/NServiceBus.Core/Encryption/DecryptBehavior.cs

```diff
@@ -9,12 +10,15 @@ namespace NServiceBus
 
     class DecryptBehavior : Behavior<LogicalMessageProcessingContext>
     {
-        EncryptionMutator messageMutator;
+        EncryptionInspector messageInspector;
+        IEncryptionService encryptionService;
 
-        public DecryptBehavior(EncryptionMutator messageMutator)
+        public DecryptBehavior(EncryptionInspector messageInspector, IEncryptionService encryptionService)
         {
-            this.messageMutator = messageMutator;
+            this.messageInspector = messageInspector;
+            this.encryptionService = encryptionService;
         }
+
         public override async Task Invoke(LogicalMessageProcessingContext context, Func<Task> next)
         {
             if (TransportMessageExtensions.IsControlMessage(context.Headers))
```

Comment:
I think we can remove this in since v6 won't invoke this part of the pipeline for control messages